retworkx-core/src/shortest_path/dijkstra.rs

```diff
@@ -20,14 +20,54 @@
 use std::collections::BinaryHeap;
 use std::hash::Hash;
 
-use indexmap::map::Entry::{Occupied, Vacant};
-
 use petgraph::algo::Measure;
-use petgraph::visit::{EdgeRef, IntoEdges, VisitMap, Visitable};
+use petgraph::graph::IndexType;
+use petgraph::visit::{EdgeRef, IntoEdges, NodeIndexable, VisitMap, Visitable};
 
 use crate::dictmap::*;
 use crate::min_scored::MinScored;
 
+// Trait for supporting storing
+pub trait DistanceMap<K, V> {
```

Comment:
Probably it's better to move this into a separate file since we might need it for other shortest path algorithms too. Also we might need to add documentation on this trait since it'll be in the docs of `retworkx-core` crate. We can even give it another name since we can in fact store any property and not just distances.